flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/partition/PartitionTable.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.taskexecutor.partition;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Thread-safe Utility for tracking partitions.
+ */
+public class PartitionTable {
+
+	private final Object lock = new Object();
+
+	private final Map<JobID, Set<ResultPartitionID>> trackedPartitionsPerJob = new HashMap<>(8);
+
+	boolean hasTrackedPartitions(JobID jobId) {
+		return trackedPartitionsPerJob.containsKey(jobId);
+	}
+
+	/**
+	 * Tracks the given partition for the given attempt.
+	 * These partitions are not considered finished.
+	 *
+	 * @param partitionId
```

Comment:
I think we could remove this @param if not comment here, otherwise should also add the `jobId` parameter.